### PR TITLE
store/tikv: remove `WithCancel` in `copIterator`.

### PR DESCRIPTION
**Pull request by coocood**
Since we already have `finished` channel to make the worker exit, we don't need to select
on `ctx.Done`, instead, we can only select `ctx.Done` in `Next`, and close the `finished` channel.

This avoid the cost to create context by `WithCancel` and select on the extra channel.

### CODE REVIEW

**Comment by coocood** [comment]
@tiancaiamao @disksing @jackysp PTAL

File: store/tikv/coprocessor.go [hunk 1]
```diff
@@ -408,18 +410,14 @@ func (it *copIterator) run(ctx goctx.Context) {
 	// Start it.concurrency number of workers to handle cop requests.
 	for i := 0; i < it.concurrency; i++ {
 		copIteratorGP.Go(func() {
-			childCtx, cancel := goctx.WithCancel(ctx)
```
Comment (tiancaiamao) [inline store/tikv/coprocessor.go]: Can we delete this directly without adding the `closed` field ?
AFAIK, the purpose of creating a new context before a new goroutine is to reduce the lock contention.
But maybe it's no problem here.

**Comment by tiancaiamao** [comment]
LGTM

File: store/tikv/coprocessor.go [hunk 2]
```diff
@@ -656,15 +656,17 @@ func calculateRemain(ranges *copRanges, split *coprocessor.KeyRange, desc bool)
 }
 
 func (it *copIterator) Close() error {
-	close(it.finished)
+	if atomic.CompareAndSwapUint32(&it.closed, 0, 1) {
```
Comment (ngaut) [inline store/tikv/coprocessor.go]: Does the iterator will be closed multiple times?

Comment (tiancaiamao) [reply store/tikv/coprocessor.go]: When `ctx.Done()` is called, `close(it.finished)` will be call here:
https://github.com/pingcap/tidb/pull/5701/files#diff-c27388ffb48c6f6eaeefe07dd3243530R443

**Pull request merged**